src/helperFunctions/hasSubArray.ts

```diff
@@ -0,0 +1,11 @@
+export const hasSubArray = (
+  master: number[][] = [[]],
```

Comment:
```suggestion
  super: number[][] = [[]],
```

Nitpicky, but if you're using `sub` the opposite is `super`. Otherwise, I think just `array` would be fine, so `array` and `subArray` etc.